src/WindowsCaptureService/WindowsCaptureService.cpp

```diff
@@ -21,17 +24,28 @@ grpc::Status WindowsCaptureService::Capture(
     grpc::ServerReaderWriter<CaptureResponse, CaptureRequest>* reader_writer) {
   orbit_base::SetCurrentThreadName("WinCS::Capture");
 
-  grpc::Status result = InitializeCapture(reader_writer);
-  if (!result.ok()) {
-    return result;
+  orbit_capture_service::ClientCaptureEventCollectorBuilderImpl
+      client_capture_event_collector_builder{reader_writer};
+
+  // shared_ptr because it might outlive this method. See wait_for_stop_capture_request_thread_ in
+  // WaitForStopCaptureRequestOrMemoryThresholdExceeded.
```

Comment:
Well, this is not the case of the Windows service. You can keep the `shared_ptr` (or if you like the factory method suggestion, make that return a `stared_ptr` directly), but that is more for consistency.

Reply:
Remove now. Thanks for pointing this out!